gripper_teleoperation:
  controller_frame_id:
    type: string
    default_value: ""
    description: "Frame ID for the 3D tracking controller"

  joystick:
    open_button:
      type: int
      default_value: -1
      description: "Button to open the gripper for binary control (-1 to disable)"
    close_button:
      type: int
      default_value: -1
      description: "Button to close the gripper for binary control (-1 to disable)"
    toggle_button:
      type: int
      default_value: -1
      description: "Button to toggle the gripper state for binary control (-1 to disable)"
    gripper_axis:
      type: int
      default_value: 0
      description: "Axis to control the gripper position for continuous control (-1 to disable)"
    invert_axis:
      type: bool
      default_value: false
      description: "Invert the joystick axis for gripper control"
    min_axis_value:
      type: double
      default_value: 0.0
      description: "Minimum value of the joystick axis for gripper control"
    max_axis_value:
      type: double
      default_value: 1.0
      description: "Maximum value of the joystick axis for gripper control"

  gripper_command_topic:
    type: string
    default_value: "~/gripper_cmd"
    description: "Topic to send gripper command actions to"

  gripper_max_effort:
    type: double
    default_value: -1.0
    description: "Max effort for the gripper controller. If negative, it will be ignored."

  joint_name:
    type: string
    default_value: "gripper"
    description: "Name of the gripper joint to control"

  min_target_value:
    type: double
    default_value: 0.0
    description: "Minimum position for the gripper in meters or radiants"

  max_target_value:
    type: double
    default_value: 0.01
    description: "Maximum position for the gripper in meters or radiants"

  publishing_rate:
    type: int
    default_value: 60
    description: "Publishing rate for the target_gripper topic in Hz (set to 0 to disable publishing)"
